pwndbg/tls.py

```diff
@@ -0,0 +1,101 @@
+"""
+Getting Thread Local Storage (TLS) information.
+"""
+import sys
+from types import ModuleType
+
+import gdb
+
+import pwndbg.arch
+import pwndbg.disasm
+import pwndbg.memory
+import pwndbg.regs
+import pwndbg.symbol
+import pwndbg.vmmap
+
+
+class module(ModuleType):
+    """Getting Thread Local Storage (TLS) information.
+    """
+    _errno_offset = None
+
+    def get_tls_base_via_errno_location(self) -> int:
+        """Heuristically determine the base address of the TLS."""
+        if pwndbg.arch.current not in ("x86-64", "i386", "arm"):
+            # Note: We doesn't implement this for aarch64 because its TPIDR_EL0 register seems always work
+            # If oneday we can't get TLS base via TPIDR_EL0, we should implement this for aarch64
```

Comment:
Ok, I can try to implement this trick for aarch64, but what's the reason `TPIDR_EL0` does not work?
Does M1/M2 still use `TPIDR_EL0` to store base address of TLS? Or does it fail just because some machines can't access it?
btw, @arcz, may you provide this libc of Apple M1/M2 aarch64 at here or the Discord server, please? Then I can debug it more easily maybe.

Reply:
Let's leave this for now. IIRC GDB just did not show that register and many others in `info regs all`.

Reply:
Ok!